spec/services/catalog/create_approval_request_spec.rb

```diff
@@ -1,114 +1,62 @@
 describe Catalog::CreateApprovalRequest, :type => :service do
-  let(:create_approval_request) { described_class.new(task) }
+  let(:subject) { described_class.new(task) }
+  let(:task) { TopologicalInventoryApiClient::Task.new(:id => "123") }
 
   around do |example|
     with_modified_env(:APPROVAL_URL => "http://localhost") do
       ManageIQ::API::Common::Request.with_request(default_request) { example.call }
     end
   end
 
-  let!(:order) { order_item.order }
-  let!(:portfolio_item) { create(:portfolio_item) }
-  let!(:order_item) { create(:order_item, :portfolio_item => portfolio_item, :topology_task_ref => "321") }
-  let(:task) { TopologicalInventoryApiClient::Task.new(:id => "321") }
+  let(:order) { order_item.order }
+  let!(:order_item) { create(:order_item, :topology_task_ref => "123") }
```

Comment:
@eclarizio is the topology_task_ref getting used for for getting the computed inventories and then again the same topology_task_ref would be used when we do the provisioning. How would we distinguish between the compute_inventory call versus provisioning call when debugging customer issues. Does it make sense to create a JSONB column that stores a hash of task refs, since we could be doing the same process for computing credentials.

Reply:
@mkanoor So, the minion is super dumb and just picks up the fact that "a task has completed" and then sends that task ID to the notify endpoint. I don't think we want to change that. Having to call topology from the minion introduces dependencies that are a headache.

However, I believe what you're suggesting would be to edit the `determine_task_relevancy.rb` file which is much simpler since we already have the topology dependencies, and we can do the order item lookup there and possibly store the data as you're suggesting.

I think that's outside the scope of this PR, though.

Reply:
@eclarizio @mkanoor is there a story around the discussion here? Is this a new scope requirement or a known one?